common_rest/src/main/java/uk/gov/dhsc/htbhf/requestcontext/aop/RequestContextAdvisor.java

```diff
@@ -0,0 +1,49 @@
+package uk.gov.dhsc.htbhf.requestcontext.aop;
+
+import lombok.RequiredArgsConstructor;
+import org.aspectj.lang.ProceedingJoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.reflect.MethodSignature;
+import uk.gov.dhsc.htbhf.requestcontext.MDCWrapper;
+import uk.gov.dhsc.htbhf.requestcontext.RequestContext;
+import uk.gov.dhsc.htbhf.requestcontext.RequestContextHolder;
+
+import java.lang.reflect.Method;
+
+import static uk.gov.dhsc.htbhf.requestcontext.MDCWrapper.REQUEST_ID_MDC_KEY;
+import static uk.gov.dhsc.htbhf.requestcontext.MDCWrapper.SESSION_ID_MDC_KEY;
+
+@RequiredArgsConstructor
+public class RequestContextAdvisor {
+
+    private final RequestContextHolder requestContextHolder;
+    private final MDCWrapper mdcWrapper;
+
+    @Around("@annotation(NewRequestContextWithSessionId)")
+    public Object updateRequestContext(ProceedingJoinPoint joinPoint) throws Throwable {
+        try {
+            String sessionId = getSessionIdFromMethodAnnotation(joinPoint);
+            String requestId = "";
```

Comment:
I think the blank string should be a constant to some some context on why it is a blank string (not just call it `BLANK_STRING`).